pkg/virtctl/imageupload/imageupload.go

```diff
@@ -926,3 +935,34 @@ func handleEventErrors(client kubecli.KubevirtClient, pvcName, dvName, namespace
 
 	return nil
 }
+
+func createDataSource(client kubecli.KubevirtClient, name, namespace string) error {
+	if dv, _ := client.CdiClient().CdiV1beta1().DataSources(namespace).Get(context.Background(), name, metav1.GetOptions{}); dv != nil {
```

Comment:
`ds` and you should check `err` here with `errors.IsAlreadyExists(err)` to ensure it already exists.

Reply:
ah `dv` just was confused in my mind, yes of course `ds` :)

Reply:
I'm not sure how `errors.IsAlreadyExists(err)` is related in Get() - Isn't usually after Create() to check the existence of the object, using the err?

For example:
```
secret, err = ctrl.Client.CoreV1().Secrets(vmExport.Namespace).Create(context.Background(), secret, metav1.CreateOptions{})
if err != nil {
	if errors.IsAlreadyExists(err) {
		return nil
	}
	return err
}
```

Reply:
Sorry yes this is a Get call so ignore my suggestion of checking if the err is AlreadyExists but we should still check err here. My point about patching an existing DS also remains.

Reply:
Thanks Lee. I've used Patch instead of Update.